Sources/Datadog/Tracing/Feature/MessageReceivers.swift

```diff
@@ -7,9 +7,28 @@
 import Foundation
 import DatadogInternal
 
-internal struct TracingMessageReceiver: FeatureMessageReceiver {
-    /// Tracks RUM context to be associated with spans.
-    let rum = TracingWithRUMIntegration()
+internal struct CoreContext {
+    /// The RUM attributes that should be added as Span tags.
+    ///
+    /// These attributes are synchronized using a read-write lock.
+    var rum: [String: String]?
+
+    /// Provides the history of app foreground / background states.
+    var applicationStateHistory: AppStateHistory?
+}
+
+internal final class ContextMessageReceiver: FeatureMessageReceiver {
```

Comment:
The `ContextMessageReceiver` replace the `TracingMessageReceiver`. It will store RUM context and Application State History received from the bus.

The RUM context will be used for span as before.
The Application State History is used for setting span tags during URLSessionTask interception.